pkg/gce-pd-csi-driver/node.go

```diff
@@ -364,7 +368,42 @@ func (ns *GCENodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRe
 }
 
 func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeGetVolumeStats is not yet implemented"))
+	if len(req.VolumeId) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "NodeGetVolumeStats volume ID was empty")
+	}
+	if len(req.VolumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "NodeGetVolumeStats volume path was empty")
+	}
+
+	exists, err := ns.Mounter.Interface.ExistsPath(req.VolumePath)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unknown error when stat on %s: %v", req.VolumePath, err)
+	}
+	if !exists {
+		return nil, status.Errorf(codes.NotFound, "path %s does not exist", req.VolumePath)
+	}
+
+	available, capacity, used, inodesFree, inodes, inodesUsed, err := ns.VolumeStatter.StatFS(req.VolumePath)
```

Comment:
what happens if this is a raw block volume?

Reply:
glad you asked. I got some weird behavior in my test:
```
"got: available 1930592256, capacity 1930592256, used 0, inodesFree 471051, inodes 471336, inodesUsed 285 -- expected: available ~= 5368709120, capacity ~= 5368709120, used = 0, inodesFree != 0, inodes != 0 , inodesUsed != 0"
```

It seems the available and capacity bytes are much smaller than expected. Also this might be just my misunderstanding but shouldn't there be `0` inodes total for an unformatted block device?

Looking into it some more but posting here just in case you have thoughts.

Reply:
ok I think these numbers are just random garbage numbers since statfs specifically gets filesystem statistics. So for a block device we would probably just want to do a different call..

Though FS/Block isn't passed into this function... we could inspect it based on the filename of the `volumePath` but that seems hacky

Reply:
cc @gnufied 

Seems like we may be missing block/fs capability in this CSI call too?  Or do we need to detect what type it is based on the path?

Reply:
These methods may be useful for you to detect if a path is a block device, along with getting the capacity of a block device:
https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/pkg/util/volume_util_linux.go

Reply:
updated to contain logic for block as well

Reply:
We could add capability here too but for now we decided not to. Depending on use case we could do this as a future update.

Also btw - used and available should not set when device is of type raw block.